### DISCUSSION TITLE
Unable to run c++ program in vscode

### DISCUSSION THREAD

**Discussion by anurawat01** [microsoft/vscode-cpptools#7501] [Q&A]
I am getting this error when I am using auto keyword in my program.

**warning: 'auto' type specifier is a C++11 extension [-Wc++11-extensions]**

Help please ?

**Comment by sean-mcmanus** [comment]
You need to add the flag "-std=c++11" when building, e.g. to "args" in tasks.json if you're building that way.

**Reply by anurawat01** [reply]
I am already using that but then also I am not able to run it.
Please help I have wasted my one full day for configuring it even I have reinstall everything then also.
tasks": [
		{
			"type": "cppbuild",
			"label": "C/C++: clang++ build active file",
			"command": "/usr/bin/clang++",
			"args": [
				"-std=c++11",
				"-stdlib=libc++",
				"${file}",
				"-o",
				"${fileDirname}/${fileBasenameNoExtension}"
			],
			"options": {
				"cwd": "${workspaceFolder}"
			},
			"problemMatcher": [
				"$gcc"
			],
			"group": {
				"kind": "build",
				"isDefault": true
			},
			"detail": "compiler: /usr/bin/clang++"
		}
	]

**Reply by sean-mcmanus** [reply]
I don't know why that isn't working. It's only a warning, so does your code compile? Does it work if you run `/usr/bin/clang++ -std=c++11 etc.` from the command line?

**Reply by anurawat01** [reply]
Yeah, If I do run this then it does work.
